### PR TITLE
feat: Service Catalog Datadog integration 

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

Adding Datadog as a new Service Catalog integration.
 

DOCU-4109
### Testing instructions

Preview link: https://deploy-preview-8154--kongdocs.netlify.app/konnect/service-catalog/integrations/datadog/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

**Comment by miko-bautista** [comment]
LGTM, thanks Diana! I'll merge the PR when we are ready for the Public Beta launch. FYI that there are a few UI polishes we'd like to address before going for an Internal Release (https://kongstrong.slack.com/archives/C068AHKGSPQ/p1731956817244099) , and so Public Beta won't happen for at least a week. Regardless, this PR is good to go from my perspective.

**Comment by miko-bautista** [comment]
Oh whoops, I accidentally clicked the "mark as ready for review" button. 😓

**Comment by cloudjumpercat** [comment]
> Oh whoops, I accidentally clicked the "mark as ready for review" button. 😓

@miko-bautista No worries! I converted it back to a draft

**Comment by cloudjumpercat** [comment]
> LGTM, thanks Diana! I'll merge the PR when we are ready for the Public Beta launch. FYI that there are a few UI polishes we'd like to address before going for an Internal Release (https://kongstrong.slack.com/archives/C068AHKGSPQ/p1731956817244099) , and so Public Beta won't happen for at least a week. Regardless, this PR is good to go from my perspective.

@miko-bautista Thanks for the review! I'd like to add a line after the integration instructions about selecting the region in Datadog (I think that's what happens here based on the brief) once I can test it in the internal release/beta. Do dashboards still need to be public for Konnect to ingest them?

**Comment by miko-bautista** [comment]
> Do dashboards still need to be public for Konnect to ingest them?

I believe Datadog Dashboards _do not_ need to be public for Service Catalog to ingest them, but I asked Miles a question on this document just to be sure: https://docs.google.com/document/d/14fNT_KPVEtgtIMlrI180jDwsHD4BaF7u5j4cqurbPKI/edit?disco=AAABY-yC8fs

File: app/konnect/service-catalog/integrations/datadog.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+mechanism: "pull/ingestion model"
+---
+
+_Type: External_
+
+The Datadog integration lets you connect Datadog entities directly to your Service Catalog services.
+
+## Authenticate the Datadog integration
+
+1. From the **{% konnect_icon servicehub %} Service Catalog** in {{site.konnect_product_name}}, select **[Integrations](https://cloud.konghq.com/us/service-catalog/integrations)**.
```
Comment (lena-larionova) [inline app/konnect/service-catalog/integrations/datadog.md:17]:
```suggestion
1. From the **{% konnect_icon servicehub %} Service Catalog** in {{site.konnect_product_name}}, select **[Integrations](https://cloud.konghq.com/service-catalog/integrations)**. 
```

**Review by lena-larionova — APPROVED** [review]
LGTM with one tiny edit.

**Review by mtjandra — APPROVED** [review]
looks good to me (from developer on konnect service catalog: [ref](https://kongstrong.slack.com/archives/C068AHKGSPQ/p1732568654018499))

**Comment by miko-bautista** [comment]
LGTM ✅

**Comment by miko-bautista** [comment]
@cloudjumpercat Whenever you get the chance, could you help me merge this PR please? We are good to go for the Public Beta launch. 🚀

**Review by miko-bautista — APPROVED** [review]

**Pull request merged**